typespec-tests/src/test/java/com/cadl/union/UnionTests.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.cadl.union;
+
+import com.azure.core.util.BinaryData;
+import com.cadl.union.models.Result;
+import com.cadl.union.models.SubResult;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+public class UnionTests {
+
+    private static final byte[] bytes = new byte[] { 'a' };
+
+    @Test
+    public void testSerialization() {
+        // test non-null Union (required property)
```

Comment:
Do we have javadoc in client method to show user how to pass in `BinaryData` (union parameter)? It could be hard for user  to create `BinaryData` object.

Reply:
Yes, we know. No time to do this yet.

Reply:
For Azure lib, we do not expect to expose this to user.

Reply:
sure, I'm fine with adding the javadoc later.